src/main/java/org/apache/commons/imaging/formats/tiff/TiffRasterDataFloat.java

```diff
@@ -0,0 +1,185 @@
+        data = new float[nCells];
+    }
+
+    /**
+     * Construct an instance allocating memory for the specified dimensions.
+     *
+     * @param width a value of 1 or greater
+     * @param height a value of 1 or greater
+     * @param data the data to be stored in the raster.
+     */
+    public TiffRasterDataFloat(final int width, final int height, final float[] data) {
+        super(width, height);
+
+        if (data == null || data.length < width * height) {
+            throw new IllegalArgumentException(
+                "Specified data does not contain sufficient elements");
+        }
+        this.data = data;
+    }
+
+     /**
+     * Gets the raster data type from the instance.
+     *
+     * @return a value of TiffRasterDataType&#46;FLOAT.
+     */
+    @Override
+    public TiffRasterDataType getDataType() {
+        return TiffRasterDataType.FLOAT;
+    }
+
+
+    /**
+     * Sets the value stored at the specified raster coordinates.
+     *
+     * @param x integer coordinate in the columnar direction
+     * @param y integer coordinate in the row direction
+     * @param value the value to be stored at the specified location;
+     * potentially a Float&#46;NaN.
+     */
+    @Override
+    public void setValue(final int x, final int y, final float value) {
+        int index = checkCoordinates(x, y);
+        data[index] = value;
+    }
+
+    /**
+     * Gets the value stored at the specified raster coordinates.
+     *
+     * @param x integer coordinate in the columnar direction
+     * @param y integer coordinate in the row direction
+     * @return the value stored at the specified location; potentially a
+     * Float&#46;NaN.
+     */
+    @Override
+    public float getValue(final int x, final int y) {
+        int index = checkCoordinates(x, y);
+        return data[index];
+    }
+
+
+    /**
+     * Sets the value stored at the specified raster coordinates.
+     *
+     * @param x integer coordinate in the columnar direction
+     * @param y integer coordinate in the row direction
+     * @param value the value to be stored at the specified location
+     */
+    @Override
+    public void setIntValue(final int x, final int y, final int value) {
+        int index = checkCoordinates(x, y);
+        data[index] = value;
+    }
+
+    /**
+     * Gets the value stored at the specified raster coordinates.
+     *
+     * @param x integer coordinate in the columnar direction
+     * @param y integer coordinate in the row direction
+     * @return the value stored at the specified location
+     */
+    @Override
+    public int getIntValue(final int x, final int y) {
+        int index = checkCoordinates(x, y);
+        return (int)data[index];
+    }
+
+
+
+
+    /**
+     * Tabulates simple statistics for the raster and returns an instance
+     * containing general metadata.
+     *
+     * @return a valid instance containing a safe copy of the current simple
+   * statistics for the raster.     */
```

Comment:
Broken Javadoc formatting (shouldn't matter for the generated javadoc I think, but better fix it here I guess.

Reply:
done